cdap-ui/app/features/services/controllers/tabs/status-ctrl.js

```diff
@@ -26,27 +26,10 @@ angular.module(PKG.name + '.feature.services')
         $scope.instances = res;
       });
 
-    dataSrc.poll({
+    dataSrc.request({
```

Comment:
One example where polling is definitely necessary is MapReduce (which stops on its own, once it completes).
I think that the UI should poll for all programs though if the user is on the page to view the status (flows/services can stop on their own given a failure condition).